git-discover/tests/isolated.rs

```diff
@@ -52,10 +52,7 @@ fn upwards_with_relative_directories_and_optional_ceiling() -> git_testtools::Re
         .unwrap_err();
 
         assert!(
-            matches!(
-                err,
-                git_discover::upwards::Error::NoGitRepositoryWithinCeiling { ceiling_height: 1, .. }
-            ),
+            matches!(err, git_discover::upwards::Error::NoMatchingCeilingDir),
```

Comment:
I'm not sure if the description ("limiting the ceiling to the CWD cannot work as it's just an empty dir") is still right.

Reply:
I have added a test which might be related. But indeed, I don't fully understand what's going on here.